src/lib/output/RunResultsProcessor.ts

```diff
@@ -1,34 +1,47 @@
 import {AnyJson} from '@salesforce/ts-types';
 import {SfError} from '@salesforce/core';
-import fs = require('fs');
 import {FormattedOutput, EngineExecutionSummary} from '../../types';
 import {BundleName, getMessage} from "../../MessageCatalog";
 import {Display} from "../Display";
 import {INTERNAL_ERROR_CODE} from "../../Constants";
-import {OutputFormat} from "../output/OutputFormat";
-import {Results} from "../output/Results";
+import {OutputFormat} from "./OutputFormat";
+import {Results} from "./Results";
+import {ResultsProcessor} from "./ResultsProcessor";
+import {writeToFile} from "./OutputUtils";
+import {JsonReturnValueHolder} from "./JsonReturnValueHolder";
 
 export type RunOutputOptions = {
 	format: OutputFormat;
+	verboseViolations: boolean;
 	severityForError?: number;
 	outfile?: string;
 }
 
-export class RunResultsProcessor {
+/**
+ * The primary ResultsProcessor used in production
+ *
+ * Note: We should consider separating this into multiple ResultProcessor classes to separate the responsibilities:
+ *       --> creating the json return value
+ *       --> creating the users outfile (where we can reuse the OutfileResultsProcessor)
+ *       --> creating console output
+ *       --> checking and throwing an exception if over severity threshold
+ *       Right now all of these are entangled with one another below and a design change would be needed to do this.
```

Comment:
Changing design and refactoring this was outside the scope of this work item. But with v4 of our cli... we might consider separating all these and detangling them from one another.